### PR TITLE
Option to download all "clapped" articles of a user

### PR DESCRIPTION
**Pull request by miry**
This way you could add a cronjob that is looking for new articles you clapped.
Like a download queue. :wink: 

Maybe there should be two variants of it:
1. naïvely overwrite everything in case it was already downloaded (i.e. `--update`?)
2. just download articles that aren't already there (would probably a better default)

### CODE REVIEW

**Comment by miry** [comment]
@alexanderadam I added ability to get articles from user's recommened list.

You can try it:

```shell
$ docker run -v /home/miry/projects/miry/medup/posts/medium_miry/claps:/posts -it miry/medup:0.1.5-alpha.1 -u <author username> --recommended

# or from git branch 2-follow-clapps:
$ crystal run src/cli.cr -- -u miry -d posts/recommended --recommended
```

Currently it always update with new version of post. The old content if it was exists, is moving to same filename with ext `old`.

I am very interesting in your feedback.

**Comment by alexanderadam** [comment]
Hi Michael,

maybe you want to use [`mkdir_p`](https://crystal-lang.org/api/master/Dir.html#mkdir_p%28path%2Cmode%3D511%29-class-method) instead of [`mkdir`](https://github.com/miry/medup/blob/master/src/medup/tool.cr#L40) because otherwise I get

```
Unhandled exception: Unable to create directory 'posts/recommended': No such file or directory (Errno)
  from /home/linuxbrew/.linuxbrew/Cellar/crystal/0.32.1/src/crystal/system/unix/dir.cr:60:7 in 'create'
  from /home/linuxbrew/.linuxbrew/Cellar/crystal/0.32.1/src/dir.cr:235:5 in 'mkdir'
  from /home/linuxbrew/.linuxbrew/Cellar/crystal/0.32.1/src/dir.cr:234:3 in 'mkdir'
  from src/medup/tool.cr:0:9 in 'save'
  from src/medup/tool.cr:25:9 in 'backup'
  from src/medup/command.cr:28:7 in 'run'
  from src/cli.cr:3:1 in '__crystal_main'
  from /home/linuxbrew/.linuxbrew/Cellar/crystal/0.32.1/src/crystal/main.cr:97:5 in 'main_user_code'
  from /home/linuxbrew/.linuxbrew/Cellar/crystal/0.32.1/src/crystal/main.cr:86:7 in 'main'
  from /home/linuxbrew/.linuxbrew/Cellar/crystal/0.32.1/src/crystal/main.cr:106:3 in 'main'
  from __libc_start_main
  from _start
  from ???
```

But other than that it seems to work _mostly_ great! :+1:

I'm getting an error on _some_ posts though:

```
GET https://medium.com/@alexanderadam/bc031f24c659?format=json => 200 OK
Unhandled exception: Unknown JSON attribute: alignment
  parsing Medium::Post::Paragraph at 1:640
  parsing Medium::PostBodyModel#paragraphs at 1:538
  parsing Medium::PostContent#bodyModel at 1:525
  parsing Medium::Post#content at 1:474 (JSON::MappingError)
  from src/medium/post.cr:0:5 in 'initialize'
  from src/medium/post.cr:8:5 in 'new'
  from /home/linuxbrew/.linuxbrew/Cellar/crystal/0.32.1/src/json/from_json.cr:13:3 in 'from_json'
  from src/medium/client/posts.cr:61:9 in 'post'
  from src/medup/tool.cr:24:16 in 'backup'
  from src/medup/command.cr:28:7 in 'run'
  from src/cli.cr:3:1 in '__crystal_main'
  from /home/linuxbrew/.linuxbrew/Cellar/crystal/0.32.1/src/crystal/main.cr:97:5 in 'main_user_code'
  from /home/linuxbrew/.linuxbrew/Cellar/crystal/0.32.1/src/crystal/main.cr:86:7 in 'main'
  from /home/linuxbrew/.linuxbrew/Cellar/crystal/0.32.1/src/crystal/main.cr:106:3 in 'main'
  from __libc_start_main
  from _start
  from ???
Caused by: Unknown JSON attribute: alignment
  parsing Medium::Post::Paragraph at 1:640
  parsing Medium::PostBodyModel#paragraphs at 1:538
  parsing Medium::PostContent#bodyModel at 1:525 (JSON::MappingError)
  from src/medium/post.cr:0:5 in 'initialize'
  from src/medium/post.cr:65:5 in 'new'
  from src/medium/post.cr:8:5 in 'initialize'
  from src/medium/post.cr:8:5 in 'new'
  from /home/linuxbrew/.linuxbrew/Cellar/crystal/0.32.1/src/json/from_json.cr:13:3 in 'from_json'
  from src/medium/client/posts.cr:61:9 in 'post'
  from src/medup/tool.cr:24:16 in 'backup'
  from src/medup/command.cr:28:7 in 'run'
  from src/cli.cr:3:1 in '__crystal_main'
  from /home/linuxbrew/.linuxbrew/Cellar/crystal/0.32.1/src/crystal/main.cr:97:5 in 'main_user_code'
  from /home/linuxbrew/.linuxbrew/Cellar/crystal/0.32.1/src/crystal/main.cr:86:7 in 'main'
  from /home/linuxbrew/.linuxbrew/Cellar/crystal/0.32.1/src/crystal/main.cr:106:3 in 'main'
  from __libc_start_main
  from _start
  from ???
Caused by: Unknown JSON attribute: alignment
  parsing Medium::Post::Paragraph at 1:640
  parsing Medium::PostBodyModel#paragraphs at 1:538 (JSON::MappingError)
  from src/medium/post.cr:0:5 in 'initialize'
  from src/medium/post.cr:79:5 in 'new'
  from src/medium/post.cr:65:5 in 'initialize'
  from src/medium/post.cr:65:5 in 'new'
  from src/medium/post.cr:8:5 in 'initialize'
  from src/medium/post.cr:8:5 in 'new'
  from /home/linuxbrew/.linuxbrew/Cellar/crystal/0.32.1/src/json/from_json.cr:13:3 in 'from_json'
  from src/medium/client/posts.cr:61:9 in 'post'
  from src/medup/tool.cr:24:16 in 'backup'
  from src/medup/command.cr:28:7 in 'run'
  from src/cli.cr:3:1 in '__crystal_main'
  from /home/linuxbrew/.linuxbrew/Cellar/crystal/0.32.1/src/crystal/main.cr:97:5 in 'main_user_code'
  from /home/linuxbrew/.linuxbrew/Cellar/crystal/0.32.1/src/crystal/main.cr:86:7 in 'main'
  from /home/linuxbrew/.linuxbrew/Cellar/crystal/0.32.1/src/crystal/main.cr:106:3 in 'main'
  from __libc_start_main
  from _start
  from ???
Caused by: Unknown JSON attribute: alignment
  parsing Medium::Post::Paragraph at 1:640 (JSON::MappingError)
  from src/medium/post/paragraph.cr:4:7 in 'initialize'
  from src/medium/post/paragraph.cr:4:7 in 'new'
  from /home/linuxbrew/.linuxbrew/Cellar/crystal/0.32.1/src/json/from_json.cr:136:11 in 'new'
  from src/medium/post.cr:79:5 in 'initialize'
  from src/medium/post.cr:79:5 in 'new'
  from src/medium/post.cr:65:5 in 'initialize'
  from src/medium/post.cr:65:5 in 'new'
  from src/medium/post.cr:8:5 in 'initialize'
  from src/medium/post.cr:8:5 in 'new'
  from /home/linuxbrew/.linuxbrew/Cellar/crystal/0.32.1/src/json/from_json.cr:13:3 in 'from_json'
  from src/medium/client/posts.cr:61:9 in 'post'
  from src/medup/tool.cr:24:16 in 'backup'
  from src/medup/command.cr:28:7 in 'run'
  from src/cli.cr:3:1 in '__crystal_main'
  from /home/linuxbrew/.linuxbrew/Cellar/crystal/0.32.1/src/crystal/main.cr:97:5 in 'main_user_code'
  from /home/linuxbrew/.linuxbrew/Cellar/crystal/0.32.1/src/crystal/main.cr:86:7 in 'main'
  from /home/linuxbrew/.linuxbrew/Cellar/crystal/0.32.1/src/crystal/main.cr:106:3 in 'main'
  from __libc_start_main
  from _start
  from ???
```

**Comment by miry** [comment]
@alexanderadam thank you for feedback and first usage. 

I have updated code to:
- Support alignment attribute for paragrpahs
- If there is an error, print message in STDERR and continue processing next posts

Also I create a seperate ticket to support `--update` option for command line.

Example of usage:

```
$ docker run -v /home/miry/projects/miry/medup/posts/medium_miry/claps:/posts -it miry/medup:0.1.5-alpha.3 -u alexanderadam --recommended

# or from git branch 2-follow-clapps:
$ crystal run src/cli.cr -- -u alexanderadam -d posts/recommended --recommended
```

P.S: You have a huge list of recommenede :+1:

**Comment by alexanderadam** [comment]
This works great :+1:

File: spec/medium/post/paragraph_spec.cr [hunk 1]
```diff
@@ -44,6 +44,11 @@ describe Medium::Post::Paragraph do
       subject.to_md.should eq("it should be cross distributive solution")
     end
 
+    it "render title" do
+      subject = Medium::Post::Paragraph.from_json(%{{"name": "d2a9", "type": 2, "text": "render tilte with picture", "markups": [], "alignment": 2}})
```
Comment (alexanderadam) [inline spec/medium/post/paragraph_spec.cr]: You probably meant `title with picture` instead of `tilte with picture` :wink:

**Comment by miry** [comment]
@alexanderadam It make sense to keep only one format. and you are right. I used JSON for debugging.

This week I will be unavailable for any work in medup. Let me create a separate ticket to choose format argument.

**Comment by alexanderadam** [comment]
> This week I will be unavailable for any work in medup. Let me create a separate ticket to choose format argument.

Easy, no rush! This is just an open source project.
Also always remember that the issues I created (or the ones that someone else will create) are just some ideas. It's nothing important nor should it be any liability.

medup is also already great the way it is! :pray:

**Pull request merged**